web/src/app/admin/keys/openai/page.tsx

```diff
@@ -49,14 +57,167 @@ const ExistingKeys = () => {
   );
 };
 
+const LLMOptions = () => {
+  const [popup, setPopup] = useState<{
+    message: string;
+    type: "success" | "error";
+  } | null>(null);
+
+  const [initialValues, setInitialValues] = useState({
+    enable_token_budget: false,
+    token_budget: "",
+    token_budget_time_period: "",
+  });
+
+  const fetchConfig = async () => {
+    const response = await fetch("/api/manage/admin/token-budget-settings");
```

Comment:
I've seen the `swr` library used elsewhere, is that a more idiomatic approach than this?

Reply:
Yea, in general we would prefer to use `swr` to standardize (although there has been some debate as to whether `swr` is the right choice 😅 )